### ISSUE TITLE
Baremetalhost Provisioning Status stuck in Inspecting State

### ISSUE THREAD

**Issue by tssavita** [metal3-io/baremetal-operator#393]
Using the following YAML file to create a baremetalhost: 
```
---
apiVersion: v1
kind: Secret
metadata:
  name: bm-secret103
type: Opaque
data:
  username: YWRtaW4=
  password: YWRtaW4=

---
apiVersion: metal3.io/v1alpha1
kind: BareMetalHost
metadata:
  name: bm-host103
spec:
  online: true
  bmc:
    address: ipmi://10.4.29.12
    credentialsName: bm-secret103
```
The target baremetalhost's network interface gets an IP from the ironic-dnsmasq container. I can see also see the PXE bootloader files being transferred through TFTP. 
From this video: https://www.youtube.com/watch?v=KIIkVD7gujY, I gathered that I need to wait for the Provisioning Status to change to *Ready*, before I can edit the baremetalhost image and checksum URLs. But the Provisioning Status is stuck in *Inspecting* State.

**Comment by zaneb** [comment]
There's no information here that we could use to debug this. I'd suggest pasting the full output of the Host CR, and the logs from the baremetal-operator container.

**Comment by tssavita** [comment]
May I ask what a Host CR? 

Here are the following are the logs from baremetal-operator.

```
{"level":"info","ts":1579627130.7716076,"logger":"cmd","msg":"Go Version: go1.10.8"}
{"level":"info","ts":1579627130.7716439,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1579627130.7716515,"logger":"cmd","msg":"Version of operator-sdk: v0.12.0"}
{"level":"info","ts":1579627130.7716596,"logger":"cmd","msg":"Component version: metal3-io/baremetal-operator was not built with version info"}
{"level":"info","ts":1579627130.771799,"logger":"cmd","msg":"gather metrics at http://127.0.0.1:8085/metrics"}
{"level":"info","ts":1579627131.128623,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":"127.0.0.1:8085"}
{"level":"info","ts":1579627131.1287663,"logger":"cmd","msg":"Registering Components."}
{"level":"info","ts":1579627131.128899,"logger":"baremetalhost_ironic","msg":"ironic settings","endpoint":"http://192.168.2.1:6385/v1/","inspectorEndpoint":"http://192.168.2.1:5050/v1/","deployKernelURL":"http://192.168.2.1:6180/images/ironic-python-agent.kernel","deployRamdiskURL":"http://192.168.2.1:6180/images/ironic-python-agent.initramfs"}
{"level":"info","ts":1579627131.1289876,"logger":"cmd","msg":"Starting the Cmd."}
{"level":"info","ts":1579627131.12922,"logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
I0121 17:18:51.129150       1 leaderelection.go:241] attempting to acquire leader lease  metal3/baremetal-operator...
I0121 17:18:51.137848       1 leaderelection.go:251] successfully acquired lease metal3/baremetal-operator
{"level":"info","ts":1579627131.1381674,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"metal3-baremetalhost-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1579627131.238593,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"metal3-baremetalhost-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1579627131.3390086,"logger":"controller-runtime.controller","msg":"Starting Controller","controller":"metal3-baremetalhost-controller"}
{"level":"info","ts":1579627131.3390584,"logger":"controller-runtime.controller","msg":"Starting workers","controller":"metal3-baremetalhost-controller","worker count":1}
{"level":"info","ts":1579636375.1360395,"logger":"baremetalhost","msg":"Reconciling BareMetalHost","Request.Namespace":"metal3","Request.Name":"bm-host103"}
{"level":"info","ts":1579636375.136138,"logger":"baremetalhost","msg":"adding finalizer","Request.Namespace":"metal3","Request.Name":"bm-host103","existingFinalizers":[],"newValue":"baremetalhost.metal3.io"}
{"level":"info","ts":1579636375.1415684,"logger":"baremetalhost","msg":"Reconciling BareMetalHost","Request.Namespace":"metal3","Request.Name":"bm-host103"}
{"level":"info","ts":1579636375.1416407,"logger":"baremetalhost","msg":"updating owner of secret","Request.Namespace":"metal3","Request.Name":"bm-host103"}
{"level":"info","ts":1579636375.1447158,"logger":"baremetalhost","msg":"changing provisioning state","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"","old":"","new":"registering"}
{"level":"info","ts":1579636375.1447685,"logger":"baremetalhost","msg":"saving host status","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"","operational status":"OK","provisioning state":"registering"}
{"level":"info","ts":1579636375.1496122,"logger":"baremetalhost","msg":"done","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"","requeue":true,"after":0}
{"level":"info","ts":1579636375.1497378,"logger":"baremetalhost","msg":"Reconciling BareMetalHost","Request.Namespace":"metal3","Request.Name":"bm-host103"}
{"level":"info","ts":1579636375.1498687,"logger":"baremetalhost","msg":"registering and validating access to management controller","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","credentials":{}}
{"level":"info","ts":1579636375.1499236,"logger":"baremetalhost","msg":"new credentials","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering"}
{"level":"info","ts":1579636375.1499445,"logger":"baremetalhost_ironic","msg":"validating management access","host":"bm-host103"}
{"level":"info","ts":1579636375.149952,"logger":"baremetalhost_ironic","msg":"looking for existing node by name","host":"bm-host103","name":"bm-host103"}
{"level":"info","ts":1579636375.2486267,"logger":"baremetalhost_ironic","msg":"registering host in ironic","host":"bm-host103"}
{"level":"info","ts":1579636375.3369336,"logger":"baremetalhost_ironic","msg":"setting provisioning id","host":"bm-host103","ID":"975c0caa-463a-41a8-afcf-a2faba069dd4"}
{"level":"info","ts":1579636375.3369775,"logger":"baremetalhost_ironic","msg":"current provision state","host":"bm-host103","lastError":"","current":"enroll","target":""}
{"level":"info","ts":1579636375.336989,"logger":"baremetalhost_ironic","msg":"changing provisioning state","host":"bm-host103","current":"enroll","existing target":"","new target":"manage"}
{"level":"info","ts":1579636375.457302,"logger":"baremetalhost","msg":"response from validate","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","provResult":{"Dirty":true,"RequeueAfter":10000000000,"ErrorMessage":""}}
{"level":"info","ts":1579636375.457453,"logger":"baremetalhost","msg":"host not ready","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","wait":10}
{"level":"info","ts":1579636375.4574678,"logger":"baremetalhost","msg":"saving host status","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","operational status":"OK","provisioning state":"registering"}
{"level":"info","ts":1579636375.4624963,"logger":"baremetalhost","msg":"publishing event","reason":"Registered","message":"Registered new host"}
{"level":"info","ts":1579636375.4668608,"logger":"baremetalhost","msg":"done","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","requeue":true,"after":10}
{"level":"info","ts":1579636375.4669383,"logger":"baremetalhost","msg":"Reconciling BareMetalHost","Request.Namespace":"metal3","Request.Name":"bm-host103"}
{"level":"info","ts":1579636375.4670475,"logger":"baremetalhost","msg":"registering and validating access to management controller","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","credentials":{"credentials":{"name":"bm-secret103","namespace":"metal3"},"credentialsVersion":"393150"}}
{"level":"info","ts":1579636375.4670796,"logger":"baremetalhost_ironic","msg":"validating management access","host":"bm-host103"}
{"level":"info","ts":1579636375.5001569,"logger":"baremetalhost_ironic","msg":"found existing node by ID","host":"bm-host103"}
{"level":"info","ts":1579636375.5001855,"logger":"baremetalhost_ironic","msg":"current provision state","host":"bm-host103","lastError":"","current":"verifying","target":"manageable"}
{"level":"info","ts":1579636375.5001986,"logger":"baremetalhost","msg":"response from validate","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","provResult":{"Dirty":true,"RequeueAfter":0,"ErrorMessage":""}}
{"level":"info","ts":1579636375.5002146,"logger":"baremetalhost","msg":"host not ready","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","wait":0}
{"level":"info","ts":1579636375.5002255,"logger":"baremetalhost","msg":"saving host status","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","operational status":"OK","provisioning state":"registering"}
{"level":"info","ts":1579636375.5056033,"logger":"baremetalhost","msg":"done","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","requeue":true,"after":0}
{"level":"info","ts":1579636375.5107658,"logger":"baremetalhost","msg":"Reconciling BareMetalHost","Request.Namespace":"metal3","Request.Name":"bm-host103"}
{"level":"info","ts":1579636375.5109205,"logger":"baremetalhost","msg":"registering and validating access to management controller","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","credentials":{"credentials":{"name":"bm-secret103","namespace":"metal3"},"credentialsVersion":"393150"}}
{"level":"info","ts":1579636375.5109756,"logger":"baremetalhost_ironic","msg":"validating management access","host":"bm-host103"}
{"level":"info","ts":1579636375.5417173,"logger":"baremetalhost_ironic","msg":"found existing node by ID","host":"bm-host103"}
{"level":"info","ts":1579636375.5417478,"logger":"baremetalhost_ironic","msg":"current provision state","host":"bm-host103","lastError":"","current":"verifying","target":"manageable"}
{"level":"info","ts":1579636375.54176,"logger":"baremetalhost","msg":"response from validate","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","provResult":{"Dirty":true,"RequeueAfter":0,"ErrorMessage":""}}
{"level":"info","ts":1579636375.5417776,"logger":"baremetalhost","msg":"host not ready","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","wait":0}
{"level":"info","ts":1579636375.5417879,"logger":"baremetalhost","msg":"saving host status","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","operational status":"OK","provisioning state":"registering"}
{"level":"info","ts":1579636375.5463233,"logger":"baremetalhost","msg":"done","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","requeue":true,"after":0}
{"level":"info","ts":1579636375.5565677,"logger":"baremetalhost","msg":"Reconciling BareMetalHost","Request.Namespace":"metal3","Request.Name":"bm-host103"}
{"level":"info","ts":1579636375.5566776,"logger":"baremetalhost","msg":"registering and validating access to management controller","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","credentials":{"credentials":{"name":"bm-secret103","namespace":"metal3"},"credentialsVersion":"393150"}}
{"level":"info","ts":1579636375.5567112,"logger":"baremetalhost_ironic","msg":"validating management access","host":"bm-host103"}
{"level":"info","ts":1579636375.6702473,"logger":"baremetalhost_ironic","msg":"found existing node by ID","host":"bm-host103"}
{"level":"info","ts":1579636375.6702845,"logger":"baremetalhost_ironic","msg":"current provision state","host":"bm-host103","lastError":"","current":"manageable","target":""}
{"level":"info","ts":1579636375.670297,"logger":"baremetalhost_ironic","msg":"have manageable host","host":"bm-host103"}
{"level":"info","ts":1579636375.6703029,"logger":"baremetalhost","msg":"response from validate","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","provResult":{"Dirty":false,"RequeueAfter":0,"ErrorMessage":""}}
{"level":"info","ts":1579636375.6703196,"logger":"baremetalhost","msg":"updating credentials success status fields","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering"}
{"level":"info","ts":1579636375.6703255,"logger":"baremetalhost","msg":"clearing previous error message","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering"}
{"level":"info","ts":1579636375.6703508,"logger":"baremetalhost","msg":"changing provisioning state","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","old":"registering","new":"inspecting"}
{"level":"info","ts":1579636375.670366,"logger":"baremetalhost","msg":"saving host status","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","operational status":"OK","provisioning state":"inspecting"}
{"level":"info","ts":1579636375.6758022,"logger":"baremetalhost","msg":"publishing event","reason":"BMCAccessValidated","message":"Verified access to BMC"}
{"level":"info","ts":1579636375.6785269,"logger":"baremetalhost","msg":"done","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"registering","requeue":true,"after":0}
{"level":"info","ts":1579636375.6785994,"logger":"baremetalhost","msg":"Reconciling BareMetalHost","Request.Namespace":"metal3","Request.Name":"bm-host103"}
{"level":"info","ts":1579636375.6787176,"logger":"baremetalhost","msg":"inspecting hardware","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"inspecting"}
{"level":"info","ts":1579636375.6787562,"logger":"baremetalhost_ironic","msg":"inspecting hardware","host":"bm-host103","status":"OK"}
{"level":"info","ts":1579636375.7074919,"logger":"baremetalhost_ironic","msg":"found existing node by ID","host":"bm-host103"}
{"level":"info","ts":1579636375.7294416,"logger":"baremetalhost_ironic","msg":"starting new hardware inspection","host":"bm-host103"}
{"level":"info","ts":1579636375.7294812,"logger":"baremetalhost_ironic","msg":"changing provisioning state","host":"bm-host103","current":"manageable","existing target":"","new target":"inspect"}
{"level":"info","ts":1579636375.8247945,"logger":"baremetalhost","msg":"saving host status","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"inspecting","operational status":"OK","provisioning state":"inspecting"}
{"level":"info","ts":1579636375.8308747,"logger":"baremetalhost","msg":"publishing event","reason":"InspectionStarted","message":"Hardware inspection started"}
{"level":"info","ts":1579636375.8340104,"logger":"baremetalhost","msg":"done","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"inspecting","requeue":true,"after":10}
{"level":"info","ts":1579636375.8341131,"logger":"baremetalhost","msg":"Reconciling BareMetalHost","Request.Namespace":"metal3","Request.Name":"bm-host103"}
{"level":"info","ts":1579636375.8342283,"logger":"baremetalhost","msg":"inspecting hardware","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"inspecting"}
{"level":"info","ts":1579636375.8342578,"logger":"baremetalhost_ironic","msg":"inspecting hardware","host":"bm-host103","status":"OK"}
{"level":"info","ts":1579636375.868795,"logger":"baremetalhost_ironic","msg":"found existing node by ID","host":"bm-host103"}
{"level":"info","ts":1579636375.8750513,"logger":"baremetalhost_ironic","msg":"starting new hardware inspection","host":"bm-host103"}
{"level":"info","ts":1579636375.8750787,"logger":"baremetalhost_ironic","msg":"changing provisioning state","host":"bm-host103","current":"inspect wait","existing target":"manageable","new target":"inspect"}
{"level":"error","ts":1579636375.8996196,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"metal3-baremetalhost-controller","request":"metal3/bm-host103","error":"action \"inspecting\" failed: hardware inspection failed: failed to change provisioning state to \"inspect\": Bad request with: [PUT http://192.168.2.1:6385/v1/nodes/975c0caa-463a-41a8-afcf-a2faba069dd4/states/provision], error message: {\"error_message\": \"{\\\"faultcode\\\": \\\"Client\\\", \\\"faultstring\\\": \\\"The requested action \\\\\\\"inspect\\\\\\\" can not be performed on node \\\\\\\"975c0caa-463a-41a8-afcf-a2faba069dd4\\\\\\\" while it is in state \\\\\\\"inspect wait\\\\\\\".\\\", \\\"debuginfo\\\": null}\"}","errorVerbose":"Bad request with: [PUT http://192.168.2.1:6385/v1/nodes/975c0caa-463a-41a8-afcf-a2faba069dd4/states/provision], error message: {\"error_message\": \"{\\\"faultcode\\\": \\\"Client\\\", \\\"faultstring\\\": \\\"The requested action \\\\\\\"inspect\\\\\\\" can not be performed on node \\\\\\\"975c0caa-463a-41a8-afcf-a2faba069dd4\\\\\\\" while it is in state \\\\\\\"inspect wait\\\\\\\".\\\", \\\"debuginfo\\\": null}\"}\nfailed to change provisioning state to \"inspect\"\ngithub.com/metal3-io/baremetal-operator/pkg/provisioner/ironic.(*ironicProvisioner).changeNodeProvisionState\n\t/go/src/github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/ironic.go:430\ngithub.com/metal3-io/baremetal-operator/pkg/provisioner/ironic.(*ironicProvisioner).InspectHardware\n\t/go/src/github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/ironic.go:588\ngithub.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost.(*ReconcileBareMetalHost).actionInspecting\n\t/go/src/github.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost/baremetalhost_controller.go:435\ngithub.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost.(*hostStateMachine).handleInspecting\n\t/go/src/github.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost/host_state_machine.go:199\ngithub.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost.(*hostStateMachine).(github.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost.handleInspecting)-fm\n\t/go/src/github.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost/host_state_machine.go:41\ngithub.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost.(*hostStateMachine).ReconcileState\n\t/go/src/github.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost/host_state_machine.go:109\ngithub.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost.(*ReconcileBareMetalHost).Reconcile\n\t/go/src/github.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost/baremetalhost_controller.go:219\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).(github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.worker)-fm\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:193\ngithub.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152\ngithub.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153\ngithub.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:2361\nhardware inspection failed\ngithub.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost.(*ReconcileBareMetalHost).actionInspecting\n\t/go/src/github.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost/baremetalhost_controller.go:437\ngithub.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost.(*hostStateMachine).handleInspecting\n\t/go/src/github.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost/host_state_machine.go:199\ngithub.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost.(*hostStateMachine).(github.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost.handleInspecting)-fm\n\t/go/src/github.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost/host_state_machine.go:41\ngithub.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost.(*hostStateMachine).ReconcileState\n\t/go/src/github.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost/host_state_machine.go:109\ngithub.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost.(*ReconcileBareMetalHost).Reconcile\n\t/go/src/github.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost/baremetalhost_controller.go:219\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).(github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.worker)-fm\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:193\ngithub.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152\ngithub.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153\ngithub.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:2361\naction \"inspecting\" failed\ngithub.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost.(*ReconcileBareMetalHost).Reconcile\n\t/go/src/github.com/metal3-io/baremetal-operator/pkg/controller/baremetalhost/baremetalhost_controller.go:223\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).(github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.worker)-fm\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:193\ngithub.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152\ngithub.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153\ngithub.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:2361","stacktrace":"github.com/metal3-io/baremetal-operator/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:258\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211\ngithub.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).(github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.worker)-fm\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:193\ngithub.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152\ngithub.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153\ngithub.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/metal3-io/baremetal-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
{"level":"info","ts":1579636376.9006224,"logger":"baremetalhost","msg":"Reconciling BareMetalHost","Request.Namespace":"metal3","Request.Name":"bm-host103"}
{"level":"info","ts":1579636376.9008226,"logger":"baremetalhost","msg":"inspecting hardware","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"inspecting"}
{"level":"info","ts":1579636376.9008472,"logger":"baremetalhost_ironic","msg":"inspecting hardware","host":"bm-host103","status":"OK"}
{"level":"info","ts":1579636376.9509773,"logger":"baremetalhost_ironic","msg":"found existing node by ID","host":"bm-host103"}
{"level":"info","ts":1579636376.959264,"logger":"baremetalhost_ironic","msg":"inspection in progress","host":"bm-host103","started_at":1579636376.0171561}
{"level":"info","ts":1579636376.959323,"logger":"baremetalhost","msg":"saving host status","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"inspecting","operational status":"OK","provisioning state":"inspecting"}
{"level":"info","ts":1579636376.9654455,"logger":"baremetalhost","msg":"done","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"inspecting","requeue":true,"after":15}
{"level":"info","ts":1579636376.9655478,"logger":"baremetalhost","msg":"Reconciling BareMetalHost","Request.Namespace":"metal3","Request.Name":"bm-host103"}
{"level":"info","ts":1579636376.9656985,"logger":"baremetalhost","msg":"inspecting hardware","Request.Namespace":"metal3","Request.Name":"bm-host103","provisioningState":"inspecting"}
```

The following is the output of ```kubectl describe baremetalhost bm-host103```
```
Name:         bm-host103
Namespace:    metal3
Labels:       <none>
Annotations:  <none>
API Version:  metal3.io/v1alpha1
Kind:         BareMetalHost
Metadata:
  Creation Timestamp:  2020-01-21T19:52:55Z
  Finalizers:
    baremetalhost.metal3.io
  Generation:        1
  Resource Version:  408542
  Self Link:         /apis/metal3.io/v1alpha1/namespaces/metal3/baremetalhosts/bm-host103
  UID:               6972cecc-addb-41c4-962a-5e9c7b4d8cdb
Spec:
  Bmc:
    Address:           ipmi://10.4.29.12
    Credentials Name:  bm-secret103
  Online:              true
Status:
  Error Message:
  Good Credentials:
    Credentials:
      Name:               bm-secret103
      Namespace:          metal3
    Credentials Version:  393150
  Hardware Profile:
  Last Updated:           2020-01-21T20:53:11Z
  Operation History:
    Deprovision:
      End:    <nil>
      Start:  <nil>
    Inspect:
      End:    <nil>
      Start:  2020-01-21T19:52:55Z
    Provision:
      End:    <nil>
      Start:  <nil>
    Register:
      End:             2020-01-21T19:52:55Z
      Start:           2020-01-21T19:52:55Z
  Operational Status:  OK
  Powered On:          false
  Provisioning:
    ID:  975c0caa-463a-41a8-afcf-a2faba069dd4
    Image:
      Checksum:
      URL:
    State:       inspecting
  Tried Credentials:
    Credentials:
      Name:               bm-secret103
      Namespace:          metal3
    Credentials Version:  393150
Events:
  Type    Reason              Age   From                         Message
  ----    ------              ----  ----                         -------
  Normal  BMCAccessValidated  60m   metal3-baremetal-controller  Verified access to BMC
  Normal  InspectionStarted   60m   metal3-baremetal-controller  Hardware inspection started
  Normal  Registered          60m   metal3-baremetal-controller  Registered new host
```

**Comment by dhellmann** [comment]
> May I ask what a Host CR is?

"CR" is custom resource and "host" is short-hand for bare metal host. You pasted the output of the BareMetalHost custom resource below, and that's part of what we need to help.

> 
> Here are the following are the logs from baremetal-operator.
> 
> ```
...
> {"level":"info","ts":1579636375.8750787,"logger":"baremetalhost_ironic","msg":"changing provisioning state","host":"bm-host103","current":"inspect wait","existing target":"manageable","new target":"inspect"}
> {"level":"error","ts":1579636375.8996196,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"metal3-baremetalhost-controller","request":"metal3/bm-host103","error":"action \"inspecting\" failed: hardware inspection failed: failed to change provisioning state to \"inspect\": Bad request with: [PUT http://192.168.2.1:6385/v1/nodes/975c0caa-463a-41a8-afcf-a2faba069dd4/states/provision], error message: {\"error_message\": \"{\\\"faultcode\\\": \\\"Client\\\", \\\"faultstring\\\": \\\"The requested action \\\\\\\"inspect\\\\\\\" can not be performed on node \\\\\\\"975c0caa-463a-41a8-afcf-a2faba069dd4\\\\\\\" while it is in state \\\\\\\"inspect wait\\\\\\\".\\\", \\\"debuginfo\\\": null}\"}","errorVerbose":"Bad request with: [PUT http://192.168.2.1:6385/v1/nodes/975c0caa-463a-41a8-afcf-a2faba069dd4/states/provision], error message: {\"error_message\": \"{\\\"faultcode\\\": \\\"Client\\\", \\\"faultstring\\\": \\\"The requested action \\\\\\\"inspect\\\\\\\" can not be performed on node \\\\\\\"975c0caa-463a-41a8-afcf-a2faba069dd4\\\\\\\" while it is in state \\\\\\\"inspect wait\\\\\\\".\\\", \\\"debuginfo\\\": null}\"}\nfailed to change provisioning state to \"inspect\"\ngithub.com/metal3-io/baremetal-operator/pkg/provisioner/ironic.
> ```

This error may be relevant. For some reason the operator is trying to re-start inspection while it is in progress. It seems the operator eventually gets past this point, though.


> The following is the output of `kubectl describe baremetalhost bm-host103`

Could you paste the output of `kubectl get baremetalhost bm-host103 -o yaml` please? That gives more detail than `describe` does.

Are there any error messages on the console of the host being provisioned?

**Comment by tssavita** [comment]
Thanks for explaining! The YAML output is as follows:
```
apiVersion: metal3.io/v1alpha1
kind: BareMetalHost
metadata:
  creationTimestamp: "2020-01-21T22:18:42Z"
  finalizers:
  - baremetalhost.metal3.io
  generation: 1
  name: bm-host103
  namespace: metal3
  resourceVersion: "430656"
  selfLink: /apis/metal3.io/v1alpha1/namespaces/metal3/baremetalhosts/bm-host103
  uid: d3b86282-6379-4ffc-b175-73a73b3bae28
spec:
  bmc:
    address: ipmi://10.4.29.12
    credentialsName: bm-secret103
  online: true
status:
  errorMessage: ""
  goodCredentials:
    credentials:
      name: bm-secret103
      namespace: metal3
    credentialsVersion: "430064"
  hardwareProfile: ""
  lastUpdated: "2020-01-21T22:20:59Z"
  operationHistory:
    deprovision:
      end: null
      start: null
    inspect:
      end: null
      start: "2020-01-21T22:18:42Z"
    provision:
      end: null
      start: null
    register:
      end: "2020-01-21T22:18:42Z"
      start: "2020-01-21T22:18:42Z"
  operationalStatus: OK
  poweredOn: false
  provisioning:
    ID: f9ef8787-0efd-4da1-8b28-fb11124ee83b
    image:
      checksum: ""
      url: ""
    state: inspecting
  triedCredentials:
    credentials:
      name: bm-secret103
      namespace: metal3
    credentialsVersion: "430064"
```
Initially, I can't see any error messages when printing ```kubectl get baremetalhosts```. But after a while, the error column shows Introspection timed out.

**Comment by dhellmann** [comment]
> Initially, I can't see any error messages when printing `kubectl get baremetalhosts`. But after a while, the error column shows Introspection timed out.

The YAML output doesn't show the error in this case, so we have to debug by looking at the logs and the console of the host.

The first thing to check is whether the host actually booted, whether it PXE booted the provisioning agent image, and then whether it reported any errors. All of that should be visible by looking at the console of the host.

What messages do you see on the console?

**Comment by tssavita** [comment]
On booting my server, one of the physical interfaces, gets an IP from the DHCP server. 
It receives the pxelinux.0 file of an ubuntu 1804 netboot image and resumes the boot process. The installation gets completed as well.. (hoping that answered your question..)

**Comment by dhellmann** [comment]
> On booting my server,

When you say "on booting" do you mean "when the server booted" or "when I booted the server"?

>  one of the physical interfaces, gets an IP from the DHCP server.
> It receives the pxelinux.0 file of an ubuntu 1804 netboot image and resumes the boot process. 

This is good.

> The installation gets completed as well.. (hoping that answered your question..)

When you say the installation completed, do you mean that the host is now running the image that you wanted provisioned to it?

**Comment by tssavita** [comment]
> When you say "on booting" do you mean "when the server booted" or "when I booted the server"?

I had to boot the server. 

> When you say the installation completed, do you mean that the host is now running the image that you wanted provisioned to it?

Yes, that's right. But the provisioning status in the output of kubectl still shows it is in the inspecting state.

**Comment by zaneb** [comment]
> This error may be relevant. For some reason the operator is trying to re-start inspection while it is in progress. It seems the operator eventually gets past this point, though.

We determine whether inspection is in progress by checking for an IntrospectionStatus record in ironic-inspector. The logs show we're checking for it almost immediately (presumably because the CR was written to?) even though we only ask to be requeued in 10s. The call to Ironic to set the new state is not synchronous, so in this case ironic-inspector has not yet created the record. We could prevent this by checking the Ironic state before starting an inspection.

When we next retry 1s later it is finding the IntrospectionStatus. There's no more data after that. Is this the end of the log? Or does it just repeat?

I can imagine a scenario in which this results in Ironic getting wedged - if it simply goes into an error state because you asked for inspection twice, but it never marks the IntrospectionStatus as failed? That seems very wrong though, as it should just reject the request to change the state to one not allowed by the state machine and not change anything internally.

Logs from the ironic and ironic-inspector containers might be the next thing to look at to try to confirm if this or something else is the cause.

**Comment by tssavita** [comment]
> There's no more data after that. Is this the end of the log? Or does it just repeat?

It repeats.. and so I trimmed it. 

Here are the last few lines from logs of ironic container: 

```
2020-01-22 08:22:57.597 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._send_sensor_data' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:23:00.278 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._check_cleanwait_timeouts' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:23:00.286 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._check_deploy_timeouts' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:23:00.292 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._check_inspect_wait_timeouts' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:23:00.295 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._check_orphan_allocations' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:23:00.296 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._check_orphan_nodes' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:23:00.300 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._check_rescuewait_timeouts' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:23:00.306 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._sync_power_states' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:23:00.309 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.drivers.modules.inspector.Inspector._periodic_check_result' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:23:00.311 27 DEBUG ironic.common.hash_ring [-] Rebuilding cached hash rings ring /usr/lib/python3.6/site-packages/ironic/common/hash_ring.py:57
2020-01-22 08:23:00.325 27 DEBUG ironic.common.hash_ring [-] Finished rebuilding hash rings, available drivers are :fake-hardware, :idrac, :ipmi, :irmc, :redfish ring /usr/lib/python3.6/site-packages/ironic/common/hash_ring.py:62
2020-01-22 08:23:00.336 27 DEBUG ironic.conductor.task_manager [req-72efb452-4406-496d-a775-15084173d69f - - - - -] Attempting to get shared lock on node f9ef8787-0efd-4da1-8b28-fb11124ee83b (for power state sync) __init__ /usr/lib/python3.6/site-packages/ironic/conductor/task_manager.py:222
2020-01-22 08:23:00.350 27 DEBUG oslo_concurrency.processutils [req-72efb452-4406-496d-a775-15084173d69f - - - - -] Running cmd (subprocess): ipmitool -I lanplus -H 10.4.29.12 -L ADMINISTRATOR -p 623 -U admin -v -R 12 -N 5 -f /tmp/tmpyy5i4fyc power status execute /usr/lib/python3.6/site-packages/oslo_concurrency/processutils.py:372
2020-01-22 08:23:00.404 27 DEBUG oslo_concurrency.processutils [req-72efb452-4406-496d-a775-15084173d69f - - - - -] CMD "ipmitool -I lanplus -H 10.4.29.12 -L ADMINISTRATOR -p 623 -U admin -v -R 12 -N 5 -f /tmp/tmpyy5i4fyc power status" returned: 0 in 0.054s execute /usr/lib/python3.6/site-packages/oslo_concurrency/processutils.py:409
2020-01-22 08:23:00.407 27 DEBUG ironic.common.utils [req-72efb452-4406-496d-a775-15084173d69f - - - - -] Execution completed, command line is "ipmitool -I lanplus -H 10.4.29.12 -L ADMINISTRATOR -p 623 -U admin -v -R 12 -N 5 -f /tmp/tmpyy5i4fyc power status" execute /usr/lib/python3.6/site-packages/ironic/common/utils.py:77
2020-01-22 08:23:00.407 27 DEBUG ironic.common.utils [req-72efb452-4406-496d-a775-15084173d69f - - - - -] Command stdout is: "Chassis Power is on
" execute /usr/lib/python3.6/site-packages/ironic/common/utils.py:78
2020-01-22 08:23:00.408 27 DEBUG ironic.common.utils [req-72efb452-4406-496d-a775-15084173d69f - - - - -] Command stderr is: "Running Get PICMG Properties my_addr 0x20, transit 0, target 0x20
Error response 0xc1 from Get PICMG Properities
Running Get VSO Capabilities my_addr 0x20, transit 0, target 0x20
Invalid completion code received: Invalid command
Discovered IPMB address 0x0
" execute /usr/lib/python3.6/site-packages/ironic/common/utils.py:79
2020-01-22 08:23:00.410 27 DEBUG ironic.conductor.task_manager [req-72efb452-4406-496d-a775-15084173d69f - - - - -] Successfully released shared lock for power state sync on node f9ef8787-0efd-4da1-8b28-fb11124ee83b (lock was held 0.07 sec) release_resources /usr/lib/python3.6/site-packages/ironic/conductor/task_manager.py:356
2020-01-22 08:23:16.385 40 DEBUG oslo_policy.policy [req-40ce7379-6d6e-42b3-8843-366ab841eed2 - - - - -] enforce: rule="is_admin" creds={"domain_id": null, "is_admin_project": true, "is_public_api": false, "project_domain_id": null, "project_id": null, "project_name": null, "roles": [], "service_project_domain_id": null, "service_project_id": null, "service_roles": [], "service_user_domain_id": null, "service_user_id": null, "system_scope": null, "user_domain_id": null, "user_id": null} target={"domain_id": null, "is_admin_project": true, "is_public_api": false, "project_domain_id": null, "project_id": null, "project_name": null, "roles": [], "service_project_domain_id": null, "service_project_id": null, "service_roles": [], "service_user_domain_id": null, "service_user_id": null, "system_scope": null, "user_domain_id": null, "user_id": null} enforce /usr/lib/python3.6/site-packages/oslo_policy/policy.py:914
2020-01-22 08:23:16.391 40 INFO eventlet.wsgi.server [req-40ce7379-6d6e-42b3-8843-366ab841eed2 - - - - -] ::ffff:127.0.0.1 "GET / HTTP/1.1" status: 200  len: 613 time: 0.0162973
2020-01-22 08:23:46.418 42 DEBUG oslo_policy.policy [req-3ec65ce1-4ab0-41f5-a9c1-a69ee040f4c4 - - - - -] enforce: rule="is_admin" creds={"domain_id": null, "is_admin_project": true, "is_public_api": false, "project_domain_id": null, "project_id": null, "project_name": null, "roles": [], "service_project_domain_id": null, "service_project_id": null, "service_roles": [], "service_user_domain_id": null, "service_user_id": null, "system_scope": null, "user_domain_id": null, "user_id": null} target={"domain_id": null, "is_admin_project": true, "is_public_api": false, "project_domain_id": null, "project_id": null, "project_name": null, "roles": [], "service_project_domain_id": null, "service_project_id": null, "service_roles": [], "service_user_domain_id": null, "service_user_id": null, "system_scope": null, "user_domain_id": null, "user_id": null} enforce /usr/lib/python3.6/site-packages/oslo_policy/policy.py:914
2020-01-22 08:23:46.422 42 INFO eventlet.wsgi.server [req-3ec65ce1-4ab0-41f5-a9c1-a69ee040f4c4 - - - - -] ::ffff:127.0.0.1 "GET / HTTP/1.1" status: 200  len: 613 time: 0.0158875
2020-01-22 08:23:52.555 51 DEBUG oslo_policy.policy [req-1fe9e25c-989f-4b71-8972-189047b104c5 - - - - -] enforce: rule="is_admin" creds={"domain_id": null, "is_admin_project": true, "is_public_api": false, "project_domain_id": null, "project_id": null, "project_name": null, "roles": [], "service_project_domain_id": null, "service_project_id": null, "service_roles": [], "service_user_domain_id": null, "service_user_id": null, "system_scope": null, "user_domain_id": null, "user_id": null} target={"domain_id": null, "is_admin_project": true, "is_public_api": false, "project_domain_id": null, "project_id": null, "project_name": null, "roles": [], "service_project_domain_id": null, "service_project_id": null, "service_roles": [], "service_user_domain_id": null, "service_user_id": null, "system_scope": null, "user_domain_id": null, "user_id": null} enforce /usr/lib/python3.6/site-packages/oslo_policy/policy.py:914
2020-01-22 08:23:52.560 51 INFO eventlet.wsgi.server [req-1fe9e25c-989f-4b71-8972-189047b104c5 - - - - -] ::ffff:192.168.2.1 "GET /v1 HTTP/1.1" status: 200  len: 1337 time: 0.0160489
2020-01-22 08:23:52.577 51 DEBUG oslo_policy.policy [req-357ba6c2-8a6c-423a-8e77-aa1624dca554 - - - - -] enforce: rule="is_admin" creds={"domain_id": null, "is_admin_project": true, "is_public_api": false, "project_domain_id": null, "project_id": null, "project_name": null, "roles": [], "service_project_domain_id": null, "service_project_id": null, "service_roles": [], "service_user_domain_id": null, "service_user_id": null, "system_scope": null, "user_domain_id": null, "user_id": null} target={"domain_id": null, "is_admin_project": true, "is_public_api": false, "project_domain_id": null, "project_id": null, "project_name": null, "roles": [], "service_project_domain_id": null, "service_project_id": null, "service_roles": [], "service_user_domain_id": null, "service_user_id": null, "system_scope": null, "user_domain_id": null, "user_id": null} enforce /usr/lib/python3.6/site-packages/oslo_policy/policy.py:914
2020-01-22 08:23:52.612 51 DEBUG oslo_policy.policy [req-357ba6c2-8a6c-423a-8e77-aa1624dca554 - - - - -] enforce: rule="show_password" creds={"domain_id": null, "is_admin_project": true, "is_public_api": false, "project_domain_id": null, "project_id": null, "project_name": null, "roles": [], "service_project_domain_id": null, "service_project_id": null, "service_roles": [], "service_user_domain_id": null, "service_user_id": null, "system_scope": null, "user_domain_id": null, "user_id": null} target={"domain_id": null, "is_admin_project": true, "is_public_api": false, "project_domain_id": null, "project_id": null, "project_name": null, "roles": [], "service_project_domain_id": null, "service_project_id": null, "service_roles": [], "service_user_domain_id": null, "service_user_id": null, "system_scope": null, "user_domain_id": null, "user_id": null} enforce /usr/lib/python3.6/site-packages/oslo_policy/policy.py:914
2020-01-22 08:23:52.613 51 DEBUG oslo_policy.policy [req-357ba6c2-8a6c-423a-8e77-aa1624dca554 - - - - -] enforce: rule="show_instance_secrets" creds={"domain_id": null, "is_admin_project": true, "is_public_api": false, "project_domain_id": null, "project_id": null, "project_name": null, "roles": [], "service_project_domain_id": null, "service_project_id": null, "service_roles": [], "service_user_domain_id": null, "service_user_id": null, "system_scope": null, "user_domain_id": null, "user_id": null} target={"domain_id": null, "is_admin_project": true, "is_public_api": false, "project_domain_id": null, "project_id": null, "project_name": null, "roles": [], "service_project_domain_id": null, "service_project_id": null, "service_roles": [], "service_user_domain_id": null, "service_user_id": null, "system_scope": null, "user_domain_id": null, "user_id": null} enforce /usr/lib/python3.6/site-packages/oslo_policy/policy.py:914
2020-01-22 08:23:52.615 51 INFO eventlet.wsgi.server [req-357ba6c2-8a6c-423a-8e77-aa1624dca554 - - - - -] ::ffff:192.168.2.1 "GET /v1/nodes HTTP/1.1" status: 200  len: 726 time: 0.0506358
2020-01-22 08:23:56.807 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._power_failure_recovery' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:23:56.809 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.drivers.modules.irmc.raid.IRMCRAID._query_raid_config_fgi_status' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:24:00.282 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._check_cleanwait_timeouts' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:24:00.289 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._check_deploy_timeouts' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:24:00.294 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._check_inspect_wait_timeouts' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:24:00.297 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._check_orphan_allocations' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:24:00.300 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._check_orphan_nodes' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:24:00.303 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._check_rescuewait_timeouts' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:24:00.307 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.conductor.manager.ConductorManager._sync_power_states' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:24:00.310 27 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic.drivers.modules.inspector.Inspector._periodic_check_result' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:24:00.315 27 DEBUG ironic.common.hash_ring [-] Rebuilding cached hash rings ring /usr/lib/python3.6/site-packages/ironic/common/hash_ring.py:57
2020-01-22 08:24:00.329 27 DEBUG ironic.common.hash_ring [-] Finished rebuilding hash rings, available drivers are :fake-hardware, :idrac, :ipmi, :irmc, :redfish ring /usr/lib/python3.6/site-packages/ironic/common/hash_ring.py:62
2020-01-22 08:24:00.339 27 DEBUG ironic.conductor.task_manager [req-72efb452-4406-496d-a775-15084173d69f - - - - -] Attempting to get shared lock on node f9ef8787-0efd-4da1-8b28-fb11124ee83b (for power state sync) __init__ /usr/lib/python3.6/site-packages/ironic/conductor/task_manager.py:222
2020-01-22 08:24:00.353 27 DEBUG oslo_concurrency.processutils [req-72efb452-4406-496d-a775-15084173d69f - - - - -] Running cmd (subprocess): ipmitool -I lanplus -H 10.4.29.12 -L ADMINISTRATOR -p 623 -U admin -v -R 12 -N 5 -f /tmp/tmpwtituwb1 power status execute /usr/lib/python3.6/site-packages/oslo_concurrency/processutils.py:372
2020-01-22 08:24:00.406 27 DEBUG oslo_concurrency.processutils [req-72efb452-4406-496d-a775-15084173d69f - - - - -] CMD "ipmitool -I lanplus -H 10.4.29.12 -L ADMINISTRATOR -p 623 -U admin -v -R 12 -N 5 -f /tmp/tmpwtituwb1 power status" returned: 0 in 0.053s execute /usr/lib/python3.6/site-packages/oslo_concurrency/processutils.py:409
2020-01-22 08:24:00.408 27 DEBUG ironic.common.utils [req-72efb452-4406-496d-a775-15084173d69f - - - - -] Execution completed, command line is "ipmitool -I lanplus -H 10.4.29.12 -L ADMINISTRATOR -p 623 -U admin -v -R 12 -N 5 -f /tmp/tmpwtituwb1 power status" execute /usr/lib/python3.6/site-packages/ironic/common/utils.py:77
2020-01-22 08:24:00.409 27 DEBUG ironic.common.utils [req-72efb452-4406-496d-a775-15084173d69f - - - - -] Command stdout is: "Chassis Power is on
" execute /usr/lib/python3.6/site-packages/ironic/common/utils.py:78
2020-01-22 08:24:00.410 27 DEBUG ironic.common.utils [req-72efb452-4406-496d-a775-15084173d69f - - - - -] Command stderr is: "Running Get PICMG Properties my_addr 0x20, transit 0, target 0x20
Error response 0xc1 from Get PICMG Properities
Running Get VSO Capabilities my_addr 0x20, transit 0, target 0x20
Invalid completion code received: Invalid command
Discovered IPMB address 0x0
" execute /usr/lib/python3.6/site-packages/ironic/common/utils.py:79
2020-01-22 08:24:00.411 27 DEBUG ironic.conductor.task_manager [req-72efb452-4406-496d-a775-15084173d69f - - - - -] Successfully released shared lock for power state sync on node f9ef8787-0efd-4da1-8b28-fb11124ee83b (lock was held 0.07 sec) release_resources /usr/lib/python3.6/site-packages/ironic/conductor/task_manager.py:356
```
and ironic-inspector: 
```
2020-01-22 08:30:52.548 1 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic_inspector.conductor.manager.periodic_clean_up' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:30:58.409 1 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic_inspector.pxe_filter.base.BaseFilter.get_periodic_sync_task.<locals>.periodic_sync_task' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:31:13.411 1 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic_inspector.pxe_filter.base.BaseFilter.get_periodic_sync_task.<locals>.periodic_sync_task' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:31:28.413 1 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic_inspector.pxe_filter.base.BaseFilter.get_periodic_sync_task.<locals>.periodic_sync_task' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:31:43.415 1 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic_inspector.pxe_filter.base.BaseFilter.get_periodic_sync_task.<locals>.periodic_sync_task' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:31:52.550 1 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic_inspector.conductor.manager.periodic_clean_up' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:31:58.418 1 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic_inspector.pxe_filter.base.BaseFilter.get_periodic_sync_task.<locals>.periodic_sync_task' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:32:13.420 1 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic_inspector.pxe_filter.base.BaseFilter.get_periodic_sync_task.<locals>.periodic_sync_task' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:32:28.422 1 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic_inspector.pxe_filter.base.BaseFilter.get_periodic_sync_task.<locals>.periodic_sync_task' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:32:43.424 1 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic_inspector.pxe_filter.base.BaseFilter.get_periodic_sync_task.<locals>.periodic_sync_task' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:32:52.552 1 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic_inspector.conductor.manager.periodic_clean_up' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:32:58.426 1 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic_inspector.pxe_filter.base.BaseFilter.get_periodic_sync_task.<locals>.periodic_sync_task' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
2020-01-22 08:33:13.427 1 DEBUG futurist.periodics [-] Submitting periodic callback 'ironic_inspector.pxe_filter.base.BaseFilter.get_periodic_sync_task.<locals>.periodic_sync_task' _process_scheduled /usr/lib/python3.6/site-packages/futurist/periodics.py:639
```
Hmm.. I must mention one other thing that I found to be a bit weird. As I understand, the target baremetal host, where I would like to provision an image, should get an IP from the ironic-dnsmasq container, only and only when I create the bm-103 baremetal custom resource. Is that right? However, in my case, the minute I reboot the target host, I guess the NIC sends out a DHCPDiscover message and it gets an offer from the dnsmasq container with an IP and hte TFTP server IP. All of this without even my creating the bm-103 baremetalhost resource in my cluster.

**Comment by maelk** [comment]
The dnsmasq DHCP service will offer an address to any node requesting it. There is no integration between dnsmasq and ironic with that regard. Dnsmasq is configured to offer an address whenver requested, and to point to the correct files for (i)pxe boot.

**Comment by zaneb** [comment]
The `Error response 0xc1 from Get PICMG Properities` makes it look like Ironic is having difficulty talking to the BMC over IPMI. @dtantsur any ideas?

**Comment by tssavita** [comment]
@maelk, ah ok. Thanks for clarifying that that is indeed how it's supposed to be.

**Comment by zaneb** [comment]
The error message has a suspicious similarity to the one in openshift-metal3/dev-scripts#895

**Comment by stbenjam** [comment]
@tssavia Were you able to verify if Ironic can talk to the BMC?

**Comment by tssavita** [comment]
Sorry, I just saw the message.. 

> When you say "on booting" do you mean "when the server booted" or "when I booted the server"? 

@dhellmann, to answer your above question, when I creat abaremetal host with the necessary IP, the PXE client reboots, without me having to manually do it. 

@stbenjam the above was possible, and so now, I think I can verify that Ironic talks to the BMC. 

There was an issue with my ironic ramdisk getting deployed. Earlier, I was using another method of PXE booting, and hadn't chainloaded the ironic ramdisk and kernel in the right way. 

I'm now at the next stage, where the baremetal host is in the "Ready" stage, and any change I make to the image and checksum are not accepted. I'll debug a bit more and create another issue for the same.. 

Thanks for the help..
